components/operations-controller/internal/director/client.go

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2020 The Compass Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package director
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/kyma-incubator/compass/components/director/pkg/graphql"
+	"github.com/kyma-incubator/compass/components/director/pkg/resource"
+	"github.com/kyma-incubator/compass/components/system-broker/pkg/types"
+	"net/http"
+)
+
+// Client defines a Director client which is capable of fetching an application
+// and notifying Director for state changes of said application
+type Client interface {
+	types.ApplicationLister
+	UpdateStatus(ctx context.Context, request Request) error
+}
+
+// defaultClient is the default implementation of the Client interface
+type defaultClient struct {
+	types.ApplicationLister
+	HTTPClient  http.Client
+	DirectorURL string
+}
+
+// TODO: Remove this struct and resuse the one from Director once the Scheduler logic is merged
+type Request struct {
```

Comment:
Reuse type defined in https://github.com/kyma-incubator/compass/pull/1731 after it is merged:
https://github.com/kyma-incubator/compass/blob/b5fa7f582b2ec7f4000948f70959293669e306d6/components/director/pkg/operation/update_operation_handler.go#L41